controller/api/public/proto_over_http_test.go

```diff
@@ -0,0 +1,411 @@
+		expectedProtoMessage := pb.Pod{
+			Name:                "some-name",
+			PodIP:               "some-name",
+			Deployment:          "some-name",
+			Status:              "some-name",
+			Added:               false,
+			ControllerNamespace: "some-name",
+			ControlPlane:        false,
+		}
+		payload, err := proto.Marshal(&expectedProtoMessage)
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		req, err := http.NewRequest(someMethod, someUrl, bytes.NewReader(payload))
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		var actualProtoMessage pb.Pod
+		err = httpRequestToProto(req, &actualProtoMessage)
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		if actualProtoMessage != expectedProtoMessage {
+			t.Fatalf("Expected request to be [%v], but got [%v]", actualProtoMessage, expectedProtoMessage)
+		}
+	})
+
+	t.Run("Given a broken request, returns http error", func(t *testing.T) {
+		var actualProtoMessage pb.Pod
+
+		req, err := http.NewRequest(someMethod, someUrl, strings.NewReader("not really protobuf"))
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		err = httpRequestToProto(req, &actualProtoMessage)
+		if err == nil {
+			t.Fatalf("Expecting error, got nothing")
+		}
+
+		if httpErr, ok := err.(httpError); ok {
+			expectedStatusCode := http.StatusBadRequest
+			if httpErr.Code != expectedStatusCode || httpErr.WrappedError == nil {
+				t.Fatalf("Expected error status to be [%d] and contain wrapper error, got status [%d] and error [%v]", expectedStatusCode, httpErr.Code, httpErr.WrappedError)
+			}
+		} else {
+			t.Fatalf("Expected error to be httpError, got: %v", err)
+		}
+	})
+}
+
+func TestWriteErrorToHttpResponse(t *testing.T) {
+	t.Run("Writes generic error correctly to response", func(t *testing.T) {
+		expectedErrorStatusCode := defaultHttpErrorStatusCode
+
+		responseWriter := newStubResponseWriter()
+		genericError := errors.New("expected generic error")
+
+		writeErrorToHttpResponse(responseWriter, genericError)
+
+		assertResponseHasProtobufContentType(t, responseWriter)
+
+		actualErrorStatusCode := responseWriter.headers.Get(errorHeader)
+		if actualErrorStatusCode != http.StatusText(expectedErrorStatusCode) {
+			t.Fatalf("Expecting response to have status code [%d], got [%s]", expectedErrorStatusCode, actualErrorStatusCode)
+		}
+
+		payloadRead, err := deserializePayloadFromReader(bufio.NewReader(bytes.NewReader(responseWriter.body.Bytes())))
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		expectedErrorPayload := pb.ApiError{Error: genericError.Error()}
+		var actualErrorPayload pb.ApiError
+		err = proto.Unmarshal(payloadRead, &actualErrorPayload)
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		if actualErrorPayload != expectedErrorPayload {
+			t.Fatalf("Expecting error to be serialised as [%v], but got [%v]", expectedErrorPayload, actualErrorPayload)
```

Comment:
Typo here: `serialised` => `serialized`. There are a few other occurrences in this file as well.

Reply:
🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺 🇬🇧 🇦🇺